CSSRE-1979: added github-action for upstream-changes

vinaybommana opened the pull request:
Hi @deepak1725
I've added the following github action to fetch the upstream changes every monday on schedule, create a new branch - upstream-changes- and raises a pull request.
based on our previous discussion, please suggest default reviewers i should add automatically for the raised PR

deepak1725 commented on .github/workflows/main.yml:
```diff
@@ -0,0 +1,58 @@
+      if: steps.sync.outputs.has_new_commits == 'true'
+      uses: actions/github-script@v6
+      with:
+        script: |
```
Can we create a standalone script file and call it here?

vinaybommana replied:
this is done

vinaybommana commented:
Hi @deepak1725

- added the simple bash script to create a upstream-changes- branch
- the github action creates a PR
- added CODEOWNERS file as the cs-sre team.
- added another github action which added given reviewers to the newly opened PR, for now I've added the suggested reviewers

please give it a look once

deepak1725 commented:
Looks good. Make sure to fix the Validation Errors visible in changelog, 
Say EOL errors and likewise.

deepak1725 reviewed (approved):
LGTM
